### PR TITLE
Automatic convex hull navigation rejection sampling

### PR DESCRIPTION
**Pull request by nkumar-bdai**
The main purpose of this PR is to enable Spot to automatically check whether any given 2D pose can be navigated to within a map. To do this, we introduce a new `allowed-regions` metadata attribute where we define a number of points that form the periphery of all valid regions within a map. We use this to precompute convex hulls of these points and then check whether a sampled navigation point is within that convex hull or not. This convex hull approach is much better than a shape-based approach because it generalizes to *any* room geometry!

This PR also updates documentation for mapping to explain how to make maps and also how to get the points needed to form the convex hull regions.

### CODE REVIEW

File: predicators/spot_utils/README.md [hunk 1]
```diff
@@ -1,5 +1,42 @@
...
+map to yield an output `.ply` pointcloud file.
+2. Install the [Open3D package](http://www.open3d.org/docs/release/getting_started.html) with `pip install open3d`.
+3. Open up a python interpreter in your terminal, and run the following commands:
+```
+import open3d as o3d
+
+pcd = o3d.io.read_point_cloud("<path to your pointcloud file>")
+o3d.visualization.draw_geometries_with_editing([pcd])
+```
+4. Within the window, do SHIFT + left click on a point to print out its 3D coords to the terminal. Copy the first two (x, y) of all the
```
Comment (tsilver-bdai) [inline predicators/spot_utils/README.md:37]: 😍

File: predicators/spot_utils/utils.py [hunk 2]
```diff
@@ -162,11 +163,16 @@ def sample_move_offset_from_target(
...
-        for cx, cy in cand_geom.vertices:
-            if cx < min_x or cy < min_y or cx > max_x or cy > max_y:
-                oob = True
+        # Check for out-of-bounds. To do this, we're looking for
+        # one allowed region where all four points defining the
+        # robot will be within the region in the new pose.
+        oob = True
+        for region in allowed_regions:
+            for cx, cy in cand_geom.vertices:
+                if region.find_simplex(np.array([cx, cy])) < 0:
```
Comment (tsilver-bdai) [inline predicators/spot_utils/utils.py:172]: I'm not familiar with this algorithm. Did you see this used for this purpose somewhere else? It sounds reasonable but maybe slow and approximate?

**Review by tsilver-bdai — APPROVED** [review]
great work!!

File: predicators/spot_utils/utils.py (hunk 2 above)
Comment (NishanthJKumar) [reply predicators/spot_utils/utils.py:172]: Got it from [here](https://stackoverflow.com/questions/16750618/whats-an-efficient-way-to-find-if-a-point-lies-in-the-convex-hull-of-a-point-cl).

I was worried about it being slow/approximate, but integration testing seems to show it's relatively fast and seems accurate enough for our purposes. I believe it gets much worse when the points are high dimensional and there are a lot of them, but for a few points in 2D, it should be pretty fine.

**Pull request merged**